tests/protocol/test_wire.py

```diff
@@ -632,7 +629,309 @@ def test_save_or_update_goal_state_should_update_goal_state_and_container_id_whe
         self.assertNotEqual(old_container_id, host.container_id)
         self.assertEquals(host.container_id, "z6d5526c-5ac2-4200-b6e2-56f2b70c5ab2")
 
-    def test_send_request_using_appropriate_channel_direct_func_works(self, *args):
+    @patch("azurelinuxagent.common.protocol.wire.WireClient.get_goal_state")
+    @patch("azurelinuxagent.common.protocol.hostplugin.HostPluginProtocol.get_artifact_request")
+    def test_download_ext_handler_pkg_should_use_retry_logic(self, mock_get_artifact_request, *args):
+        mock_get_artifact_request.return_value = "dummy_url", "dummy_header"
+        protocol = WireProtocol("foo.bar")
+        HostPluginProtocol.set_default_channel(False)
+
+        mock_failed_response = MockResponse(body=b"", status_code=httpclient.GONE)
+        mock_successful_response = MockResponse(body=b"OK", status_code=200)
+        destination = os.path.join(self.tmp_dir, "tmp_file")
+
+        original_direct = protocol.client.stream
+        original_host = protocol.download_ext_handler_pkg_through_host
+
+        def mock_direct(*args, **kwargs):
```

Comment:
let's try using wraps instead of the helpers:

> with patch("azurelinuxagent.common.protocol.wire.WireClient.stream", wraps=protocol.client.stream) as patch_direct:

Reply:
Thanks, didn't know about this.